Kudu.Tests/Core/Function/KafkaTriggerKedaAuthProviderTest.cs

```diff
@@ -32,17 +49,67 @@ public void PopulateAuthenticationRef_Fails_When_TriggerAuthCreationFails()
             KafkaTriggerKedaAuthProviderErrorMock kafkaTriggerKedaAuthProvider = new KafkaTriggerKedaAuthProviderErrorMock();
             string jsonText = @"
             {
-                ""Protocol"": ""SASL_SSL"",
-                ""AuthenticationMode"": ""PLAINTEXT"",
-                ""Username"": ""test"",
-                ""Password"": ""test""
+                ""protocol"": ""SaslSsl"",
+                ""authenticationMode"": ""Plain"",
+                ""username"": ""test"",
+                ""password"": ""test""
             }";
 
             JToken jsonObj = JToken.Parse(jsonText);
             IDictionary<string, string> authRef = kafkaTriggerKedaAuthProvider.PopulateAuthenticationRef(jsonObj, "testFunctionName");
             Assert.Null(authRef);
         }
 
+        [Fact]
+        public void PopulateAuthenticationRef_Continues_When_AddSecretsFails()
+        {
+            KafkaTriggerKedaAuthProviderOverload kafkaTriggerKedaAuthProvider = new KafkaTriggerKedaAuthProviderOverload();
+            string jsonText = @"
+            {
+                ""protocol"": ""SaslSsl"",
+                ""authenticationMode"": ""Plain"",
+                ""username"": ""test"",
+                ""password"": ""test""
+            }";
+
+            JToken jsonObj = JToken.Parse(jsonText);
+            IDictionary<string, string> authRef = kafkaTriggerKedaAuthProvider.PopulateAuthenticationRef(jsonObj, "testFunctionName");
+            Assert.Equal(1, authRef.Count);
+        }
+
+        [Fact]
+        public void TestIFTriggerAuthIsNull_With_NoAuthenticationMode()
+        {
+            KafkaTriggerKedaAuthProviderOverload kafkaTriggerKedaAuthProvider = new KafkaTriggerKedaAuthProviderOverload();
+            string jsonText = @"
+            {
+                ""protocol"": ""SaslSsl"",
+                ""username"": ""test"",
+                ""password"": ""test""
+            }";
+
+            JToken jsonObj = JToken.Parse(jsonText);
+            IDictionary<string, string> authRef = kafkaTriggerKedaAuthProvider.PopulateAuthenticationRef(jsonObj, "testFunctionName");
+            Assert.Null(authRef);
+        }
+
+        [Fact]
+         public void TestIFTriggerAuthIsNull_With_NoProtocol()
+        {
+            KafkaTriggerKedaAuthProviderOverload kafkaTriggerKedaAuthProvider = new KafkaTriggerKedaAuthProviderOverload();
+            string jsonText = @"
+            {
+                ""authenticationMode"": ""Plain"",
+                ""username"": ""test"",
+                ""password"": ""test""
+            }";
+
+            JToken jsonObj = JToken.Parse(jsonText);
+            IDictionary<string, string> authRef = kafkaTriggerKedaAuthProvider.PopulateAuthenticationRef(jsonObj, "testFunctionName");
+            Assert.Null(authRef);
+        }
+
+
```

Comment:
Don't we need the mock and validation for AddTriggerAuthAppSettingsSecrets ?

Reply:
we have KafkaTriggerKedaAuthProviderErrorMock which throws exception for AddTriggerAuthAppSettingsSecrets and is used for testing the case PopulateAuthenticationRef_Continues_When_AddSecretsFails()